processing/src/main/java/org/apache/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -265,11 +306,12 @@ private void doAggregate(
   {
     rowContainer.set(row);
 
+    long totalIncrementalBytes = 0L;
     for (int i = 0; i < aggs.length; i++) {
       final Aggregator agg = aggs[i];
       synchronized (agg) {
         try {
-          agg.aggregate();
+          totalIncrementalBytes += agg.aggregateWithSize();
```

Comment:
This is going to call the `WithSize` version even when it's not supposed to (when it's using the max size).  Please adjust to only call the `WithSize` version when the flag is set.

Reply:
Fixed.